examples/sprites/main.rs

```diff
@@ -0,0 +1,279 @@
+            sprite::SpriteSheetDefinition::new(sprite_w, sprite_h, 2, 6, false);
+
+        let sprite_sheet_index = 0;
+        let sprite_sheet = sprite_sheet_loader::load(sprite_sheet_index, &sprite_sheet_definition);
+
+        let sprite_sheet_material = {
+            let mat_defaults = world.read_resource::<MaterialDefaults>();
+            Material {
+                albedo: sprite_sheet_texture.clone(),
+                ..mat_defaults.0.clone()
+            }
+        };
+
+        // Load animations
+        let grey_bat_animation = animation::grey_bat(&sprite_sheet, &mut world);
+        let brown_bat_animation = animation::brown_bat(&sprite_sheet, &mut world);
+
+        // Calculate offset to centre all sprites
+        //
+        // The X offset needs to be multiplied because we are drawing the sprites across the window;
+        // we don't need to multiply the Y offset because we are only drawing the sprites in 1 row.
+        let sprite_count = sprite_sheet.sprites.len();
+        let sprite_offset_x = sprite_count as f32 * sprite_w / 2.;
+        let sprite_offset_y = sprite_h / 2.;
+
+        let (width, height) = {
+            let dim = world.read_resource::<ScreenDimensions>();
+            (dim.width(), dim.height())
+        };
+        // This `Transform` moves the sprites to the middle of the window
+        let mut common_transform = Transform::default();
+        common_transform.translation = Vector3::new(
+            width / 2. - sprite_offset_x,
+            height / 2. - sprite_offset_y,
+            0.,
+        );
+
+        // Store sprite sheet texture in the world's `MaterialTextureSet` resource (singleton hash
+        // map)
+        world
+            .write_resource::<MaterialTextureSet>()
+            .insert(sprite_sheet_index, sprite_sheet_texture);
+
+        // Create an entity per sprite.
+        for i in 0..sprite_count {
+            let mut sprite_transform = Transform::default();
+            sprite_transform.translation = Vector3::new(i as f32 * sprite_w, 0., 0.);
+
+            // This combines multiple `Transform`ations.
+            // You need to `use amethyst::core::cgmath::Transform`;
+            sprite_transform.concat_self(&common_transform);
+
+            let mesh = {
+                let loader = world.read_resource::<Loader>();
+                loader.load_from_data(
+                    create_mesh_vertices(sprite_w, sprite_h).into(),
+                    (),
+                    &world.read_resource::<AssetStorage<Mesh>>(),
+                )
+            };
+
+            let animation = if i < (sprite_count >> 1) {
+                grey_bat_animation.clone()
+            } else {
+                brown_bat_animation.clone()
+            };
+
+            let entity = world
+                .create_entity()
+                // The default `Material`, whose textures will be swapped based on the animation.
+                .with(sprite_sheet_material.clone())
```

Comment:
Is this clone necessary?

Reply:
Yeap, because the loop creates multiple entities, and each of them have their own material component

Reply:
I didn't realized it was a loop